docs/junit-extensions.md

```diff
@@ -34,6 +66,88 @@ The [ClearInterruptions](../junit-extensions/src/main/java/org/codice/junit/rule
   }
 ```
 
+#### MethodRuleChain
+The [MethodRuleChain](../junit-extensions/src/main/java/org/codice/junit/rules/MethodRuleChain.java) provides a JUnit4 rule that can be used to create a controlled chain of method rules when the order they are processed is important.
+
+```
+  public class MyTest {
+    @Rule public final MethodRuleChain chain = 
+      MethodRuleChain
+        .outer(new RestoreSystemProperties())
+        .around(new ClearInterruptions());
+
+    @Test
+    public void testSomething() throws Exception {
+      System.setProperty("ddf.home", "some new location");
+      final MyClass obj = new MyClass(some);
+      
+      obj.doSomething();
+    }
+  }
+```
+
+### JUnit Method Rules
+
+#### RestoreSystemProperties
+The [RestoreSystemProperties](../junit-extensions/src/main/java/org/codice/junit/RestoreSystemProperties.java) annotation can be used in conjunction with the MethodRuleAnnotationProcessor JUnit method rule to indicate all methods of a test class or specific ones where system properties should automatically be reset to their initial values after testing.
+
+#### ClearInterruptions
+The [ClearInterruptions](../junit-extensions/src/main/java/org/codice/junit/ClearInterruptions.java) annotation can be used in conjunction with the MethodRuleAnnotationProcessor JUnit method rule to indicate all methods of a test class or specific ones where the interruption state from the current thread should automatically be cleared after testing.
+
+### JUnit Test Runners
+
+#### MethodRuleAnnotationProcessor
+The [MethodRuleAnnotationRunner](../junit-extensions/src/main/java/org/codice/junit/MethodRuleAnnotationRunner.java) provides a JUnit4 test runner that supports annotation-based method rules similar support to Spock extensions by allowing specification of simple JUnit method rules via the meta-annotation [ExtensionMethodRuleAnnotation](../junit-extensions/src/main/java/org/codice/junit/ExtensionMethodRuleAnnotation.java).
```

Comment:
When we move to JUnit 5, we can easily adapt our extensions to the `@ExtendWith` mechanism. At least those that don't require any customizations. For now, I will keep those as is since this implementation allows the ability to actually customize the extension via the annotation which will be useful for subsequent annotations I am preparing for config admin.